### PR TITLE
New feature: Execute query on db profiler

### PR DESCRIPTION
**Pull request by sergio-toro**
My coworker (@AMoQu) and I have written a little feature to execute logged queries on the db profiler.

### CODE REVIEW

File: Controller/ProfilerController.php [hunk 1]
```diff
@@ -35,6 +35,49 @@ class ProfilerController extends ContainerAware
...
+        $profile = $profiler->loadProfile($token);
+        $queries = $profile->getCollector('db')->getQueries();
+
+        if (!isset($queries[$connectionName][$query])) {
+            return new Response('This query does not exist.');
+        }
+
+        $query = $queries[$connectionName][$query];
+        if (!$query['explainable']) {
+            return new Response('This query cannot be explained.');
```
Comment (stof) [inline Controller/ProfilerController.php]: the message is wrong, as this is not about explaining

**Comment by stof** [comment]
this allows to replay INSERT, DELETE and UPDATE queries, which is an issue

File: Controller/ProfilerController.php (hunk 1 above)
Comment (AMoQu) [inline Controller/ProfilerController.php]: Do you think we should remove the if condition or just change message text to something like 'This query cannot be executed.'

**Comment by AMoQu** [comment]
@stof We have patched it to only allow SELECT queries. It is okay now?

**Comment by stof** [comment]
The issue is that this does not allow all select queries anymore (not all select queries have the SELECT at the beginning, for instance when using CTEs in PostgreSQL)

**Comment by stof** [comment]
I'm -1 on this. Making it possible to re-execute queries is dangerous as it is quite impossible to figure out whether the query is safe or no to re-execute.

**Comment by sergio-toro** [comment]
@stof Yep, I know it could be dangerous, but for our use case it just saves tons of work for debugging queries because we have more than 10 different connections.
Please just give it a chance and think about it.

**Comment by guilhermeblanco** [comment]
-1 on this... closing

**Pull request closed without merging**